pkg/route/registry/route/strategy.go

```diff
@@ -214,14 +214,37 @@ func (s routeStrategy) validateHostUpdate(ctx apirequest.Context, route, older *
 		if hostChanged {
```

Comment:
this isn't quite what was described in https://github.com/openshift/origin/pull/18177#issuecomment-360660024:
> 1. check update if the host has changed, and if false, disallow any changes
> 2. if host hasn't changed but certs have, check "create", and if false, disallow changes
> 3. allow changes

I think we should only run the `update` `custom-host` SAR (the one above) if `hostChanged`

Reply:
@liggitt It implements the logic as described. Although this code does an extra check for 'update' always, it is to prevent a case where someone has an 'update' permission but does not have 'create'. We should allow that case to be able to update TLS certs.
Review again please.